Lib/idlelib/pyshell.py

```diff
@@ -929,6 +937,33 @@ def __init__(self, flist=None):
         #
         self.pollinterval = 50  # millisec
 
+        self.shell_sidebar = self.ShellSidebar(self)
+        self.shell_sidebar.show_sidebar()
+
+        # Insert UserInputTaggingDelegator at the top of the percolator,
+        # but make calls to text.insert() skip it.  This causes only insert
+        # events generated in Tcl/Tk to go through this delegator.
+        self.text.insert = self.per.top.insert
+        self.per.insertfilter(UserInputTaggingDelegator())
+
+    def get_tag_colors(self):
+        tag_colors = super().get_tag_colors()
+
+        theme = idleConf.CurrentTheme()
+        tag_colors.update({
+            "stdin": {'background': None, 'foreground': None},
+            "stdout": idleConf.GetHighlight(theme, "stdout"),
+            "stderr": idleConf.GetHighlight(theme, "stderr"),
+            "console": idleConf.GetHighlight(theme, "console"),
+        })
```

Comment:
You mean overriding the parent class's method rather than extending it? Isn't that a recipe for future bugs in later changes to `EditorWindow` or other classes in the hierarchy?